src/Query/TaskGroup.ts

```diff
@@ -1,25 +1,31 @@
 import type { Task } from '../Task';
-import type { GroupHeading } from './GroupHeading';
+import type { GroupDisplayHeading } from './GroupDisplayHeading';
 
 /**
- * TaskGroup stores a single group of tasks, that all share the same group names.
- * TaskGroup objects are stored in a TaskGroups object.
+ * Store a single group of tasks, that all share the same group names.
+ * {@link TaskGroup} objects are stored in a {@link TaskGroups} object.
  *
  * For example, if the user supplied these 'group by' lines:
  *   group by folder
  *   group by filename
  *   group by heading
- * Then the names of one TaskGroup might be this:
+ * Then the names of one {@link TaskGroup} might be this:
  *   Some/Folder/In/The/Vault
  *   A Particular File Name
  *   My lovely heading
- * And the TaskGroup would store all the tasks from that location
+ * And the {@link TaskGroup} would store all the tasks from that location
  * that match the task block's filters, in the task block's sort order
  */
 export class TaskGroup {
     /**
-     * The names of the group properties for this set of tasks,
-     * in the order of the 'group by' lines the user specified
+     * The full list of names of the group properties for this set of tasks,
+     * in the order of the 'group by' lines the user specified.
+     *
+     * If there were no 'group by' instructions in the tasks code block,
+     * this will be empty.
+     *
+     * Note that the strings returned contain markdown for rendering,
+     * if required.
```

Comment:
for a query
```
group by due
group by folder
```

this will be `['due', 'folder']`? regardless of the tasks, filtered by the query?

Reply:
then these a probably `GroupDisplayHeading.name`s? =)